### PR TITLE
Fixes issue #47 : Backend Integration

### PR DESCRIPTION
**Pull request by AmanSingh494**
Fixes issue #47 
### Problem
DecenTrade lacked backend for secure requests to IPFS and database

### Solution
Creating a backend

### Changes
1. Added Backend folder
2. Created Server
3. Added IPFS interaction routes
4. Updated frontend to interact with backend

### CODE REVIEW

**Review by 4darsh-Dev — COMMENTED** [review]
ensure to use branch name other than main
it create confusing while resolving merge conflicts

File: frontend/src/utils/ethereum.js
Comment (4darsh-Dev) [inline frontend/src/utils/ethereum.js:1]: Audit: Unsanitized user input passed to server logs JS-A1004
Security
```html
Bad Practice

import http from "http"
import url from "url"

http.createServer((req, res) => {
  const parsedUrl = url.parse(req.url, true)
  // Vulnerable! user can inject special characters in the terminal
  console.log(parsedUrl.query.username);
})

Recommended

import http from "http"
import url from "url"

http.createServer((req, res) => {
  const parsedUrl = url.parse(req.url, true)

  // NOTE: Ideally, stronger sanitization functions should be used.
  // String#replace is only used as an example.
  const username = parsedUrl.query.username.replace(/\n|\r/g, "")
  console.log(parsedUrl.username);
})


```

File: frontend/src/utils/ethereum.js [hunk 1]
```diff
@@ -1,87 +1,133 @@
...
+            body: formData,
+        })
+        const data = await response.json()
+        console.log('IPFS Response:', data)
+        return data.ipfsHash
+    } catch (e) {
+        console.error(e)
+    }
+}
+const uploadMetadataToIPFS = async (metadata) => {
```
Comment (4darsh-Dev) [inline frontend/src/utils/ethereum.js:64]: Any code paths that do not have explicit returns will return undefined. It is recommended to replace any implicit dead-ends that return undefined with a return null statement.
```html
Bad Practice

function getUser(name) {
    if (userExists(name)) {
        return userDb.get(name);
    }
    // whoops! Implicit undefined return
}

function readFile(fileName) {
    if (fs.existsSync(fileName)) {
        return fs.readFileSync(fileName, 'utf-8');
    }

    return; // Implicit undefined return.
}

Recommended

function getUser(name) {
    if (userExists(name)) {
        return userDb.get(name);
    }

    // `null` indicates there is no such user.
    return null;
}

function readFile(fileName) {
    if (fs.existsSync(fileName)) {
        return fs.readFileSync(fileName, 'utf-8');
    }

    // `null` indicates there is no such file.
    return null;
}



```

**Review by 4darsh-Dev — APPROVED** [review]
Nice work @AmanSingh494 , 
I've suggested some possible improvements 
Go through them and Follow best practices 
Try to  fix them in your next PR

**Comment by AmanSingh494** [comment]
Thanks @4darsh-Dev for giving me the chance to work on this issue and I will certainly follow the best practices in future prs.

**Pull request merged**